contrib/golang/filters/http/source/go/pkg/http/shim.go

```diff
@@ -276,16 +284,97 @@ func envoyGoFilterOnHttpLog(r *C.httpRequest, logType uint64) {
 
 	v := api.AccessLogType(logType)
 
+	// Request headers must exist because the HTTP filter won't be run if the headers are
+	// not sent yet.
+	reqHeader := &requestHeaderMapImpl{
```

Comment:
better add a TODO commet: it's better to make it readonly?

Reply:
Updated.